tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -3945,17 +3946,43 @@ port::Status CudnnSupport::DoPrepareForCtcLoss(
       static_cast<const CudnnRnnStateTensorDescriptor&>(probs_desc);
   const CudnnRnnStateTensorDescriptor& cudnn_grads_desc =
       static_cast<const CudnnRnnStateTensorDescriptor&>(grads_desc);
-  RETURN_IF_CUDNN_ERROR(cudnnGetCTCLossWorkspaceSize(
-      /*handle=*/cudnn.handle(), /*probsDesc=*/cudnn_probs_desc.handle(),
-      /*gradientsDesc=*/cudnn_grads_desc.handle(),
-      /*labels=*/labels_data.data(),
-      /*labelLengths=*/labels_lengths_data.data(),
-      /*inputLengths=*/input_lengths_data.data(),
-      /*algo=*/
+
+  // When determinism is required, we will only try the deterministic algo.
```

Comment:
JFYI you could do this as:

```
absl::InlinedVector<int, 2> eligible_algorithms;
eligible_algorithms.push_back(CUDNN_CTC_LOSS_ALGO_DETERMINISTIC);
if (!RequireCudnnDeterminism()) {
  eligible_algorithms.push_back(CUDNN_CTC_LOSS_ALGO_NON_DETERMINISTIC);
}

for (int algo : eligible_algorithms) {
  // Try running with `algo`, if successful then pick it.  We can order eligible_algorithms so that the first non deterministic algorithm is first and thus preferentially picked.
}
```